service-base/src/main/java/org/eclipse/hono/service/tenant/BaseTenantService.java

```diff
@@ -143,30 +150,41 @@ public final void processTenantMessage(final Message<JsonObject> tenantMsg) {
                     break;
                 }
                 payload = getRequestPayload(body);
-                log.debug("creating tenant [{}] with data {}", tenantId, payload != null ? payload.encode() : null);
+                log.debug("creating tenant [{}]", tenantId);
                 add(tenantId, payload, result -> reply(tenantMsg, result));
                 break;
             case ACTION_UPDATE:
                 payload = getRequestPayload(body);
-                log.debug("updating tenant [{}] with data {}", tenantId, payload != null ? payload.encode() : null);
+                log.debug("updating tenant [{}]", tenantId);
                 update(tenantId, payload, result -> reply(tenantMsg, result));
                 break;
             case ACTION_REMOVE:
                 log.debug("deleting tenant [{}]", tenantId);
                 remove(tenantId, result -> reply(tenantMsg, result));
                 break;
-            default:
-                break;
             }
+        } catch (final IllegalArgumentException e) {
+            log.debug("invalid action in request message [{}]", e.getMessage());
+            reply(tenantMsg, TenantResult.from(HTTP_BAD_REQUEST));
         } catch (final ClassCastException e) {
             log.debug("malformed request message [{}]", e.getMessage());
             reply(tenantMsg, TenantResult.from(HTTP_BAD_REQUEST));
+        } catch (final NullPointerException e) {
+            log.debug("malformed request message [{}]", e.getMessage());
+            if (tenantMsg != null) {
+                reply(tenantMsg, TenantResult.from(HTTP_BAD_REQUEST));
+            }
         }
     }
 
     private void reply(final Message<JsonObject> request, final AsyncResult<TenantResult> result) {
         if (result.succeeded()) {
-            this.reply(request, result.result());
+            if (result != null) {
```

Comment:
I think you mean `result.result()`. Please choose another name for the async result param